ui/app/js/controllers/release_scheduled_change_edit_controller.js

```diff
@@ -0,0 +1,144 @@
+
+angular.module('app').controller('EditReleaseScheduledChangeCtrl',
+function ($scope, $modalInstance, CSRF, Releases, sc) {
+
+  $scope.is_edit = true;
+  $scope.original_sc = sc;
+  $scope.sc = angular.copy(sc);
+  $scope.products = [];
+  Releases.getProducts().success(function(response) {
+    $scope.products = response.product;
+  });
+
+  $scope.errors = {};
+  $scope.saving = false;
+
+  $scope.setWhen = function(newDate) {
+    if (!newDate) {
+      newDate = new Date($("#id_when")[0].value);
+      $scope.sc.when = newDate;
+    }
+    $scope.calendar_is_open = false;
+    if (newDate <= new Date()) {
+      $scope.errors.when = ["Scheduled time cannot be in the past"];
+      $scope.sc.when = $scope.original_sc.when;
+    }
+    else {
+      $scope.errors.when = null;
+    }
+  };
+
+  $scope.clearWhen = function () {
+    $scope.sc.when = null;
+    $scope.errors.when = null;
+  };
+
+  $scope.fillName = function () {
+    var file = $scope.dataFile;
+    $scope.errors.data = [];
+    var reader = new FileReader();
+    reader.onloadend = function(evt) {
+      var blob = evt.target.result;
+      $scope.$apply( function() {
+        try {
+          var name = JSON.parse(blob).name;
+          if(!name) {
+            $scope.errors.data = ["Form submission error", "Name missing in blob.\n"];
+          }
+          else if (name !== $scope.sc.name) {
+            $scope.errors.data = ["Form submission error", "Name differs compared to name in blob.\n"];
+          }
+        }catch(err) {
+          $scope.errors.data = ["Form submission error", "Malformed JSON file.\n"];
+        }
+      });
+    };
+    if (typeof file !== 'undefined') {
+      // should work
+      reader.readAsText(file);
+    }
+
+  };
+
+  $scope.changeName = function () {
+    //wait for actual file to be loaded
+    setTimeout($scope.fillName, 0);
+  };
+
+  $scope.saveChanges = function () {
+    if (!$scope.sc.product.trim()) {
```

Comment:
I have created release_scheduled_change_delete_controller.js  for Scheduled Deletion